### ISSUE TITLE
Rubocop config "include" is not absolute

### ISSUE THREAD

**Issue by r4do** [rubocop/rubocop#14361]
Hey folks,
When `Include` path matches app parent directories, this might lead to false positives. An example: rubocop-rails has such piece in the [config](https://github.com/rubocop/rubocop-rails/blob/master/config/default.yml#L753-L763):
```
  Include:
    - '**/app/**/*.rb'
```
When the path of entire application includes `/app/`, this means that all ruby files will be scanned, which I believe is not how this config is designed.
In our specific case we have CI build run in Docker in `/app`, so all ruby files are kinda included and we have green rubocop output on the local, and red on CI.

I see `make_excludes_absolute` methods in `lib/rubocop/config.rb`, should it be applied to includes as well? If you guys think it's the way to fix it, I can submit the PR with the fix.

--------

## Expected behavior

Rubocop includes files by the mask checking only application files.

## Actual behavior

Rubocop includes files by the mask checking absolute path
Describe here what actually happened.
Please use `rubocop --debug` when pasting rubocop output as it contains additional information.

## Steps to reproduce the problem

1. Move an application to a `/app/` sub-directory. E.g. if absolute path of the file previously was /applications/config/application.rb, it should be now /applications/**app**/config/application.rb

2. In any cop config set includes to `**/app/**/*.rb`:
```
Rails/Exit:
  Include:
    - '**/app/**/*.rb'
```

3. Put an offence to `config/application.rb` file:
```
  exit(1)
```

4. Run rubocop and see an offence

## RuboCop version

```
❯ rubocop -V
1.78.0 (using Parser 3.3.8.0, rubocop-ast 1.45.1, analyzing as Ruby 3.2, running on ruby 3.3.4) [arm64-darwin24]
  - rubocop-rails 2.32.0
  - rubocop-rspec 3.6.0
  - rubocop-performance 1.25.0
```
